sdk/core/core/test/cmocka/test_az_policy.c

```diff
@@ -97,6 +97,96 @@ void test_az_http_pipeline_policy_telemetry(void** state)
   assert_return_code(az_http_pipeline_policy_telemetry(policies, &telemetry, &hrb, NULL), AZ_OK);
 }
 
+static void test_az_http_pipeline_policy_header_validation(void** state)
+{
+  (void)state;
+
+  uint8_t buf[100];
+  uint8_t header_buf[(2 * sizeof(az_pair))];
+  memset(buf, 0, sizeof(buf));
+  memset(header_buf, 0, sizeof(header_buf));
+
+  az_span url_span = AZ_SPAN_FROM_BUFFER(buf);
+  az_span remainder = az_span_copy(url_span, AZ_SPAN_FROM_STR("url"));
+  assert_int_equal(az_span_size(remainder), 97);
+  az_span header_span = AZ_SPAN_FROM_BUFFER(header_buf);
+  _az_http_request hrb;
+
+  assert_return_code(
+      az_http_request_init(
+          &hrb, &az_context_app, az_http_method_get(), url_span, 3, header_span, AZ_SPAN_NULL),
+      AZ_OK);
+
+  _az_http_policy policies[1] = {            
+            {
+              ._internal = {
+                .process = test_policy_transport,
+                .p_options = NULL,
+              },
+            },
+        };
+  az_http_response response = { 0 };
+  assert_return_code(
+      az_http_response_init(
+          &response,
+          AZ_SPAN_FROM_STR("HTTP/1.1 404 Not Found\r\n"
+                           "Header11: Value11\r\n"
+                           "Header22: NNNNOOOOPPPPQQQQRRRRSSSSTTTTUUUUVVVVWWWWXXXXYYYYZZZZ\r\n"
+                           "Header33:\r\n"
+                           "Header44: cba888888777777666666555555444444333333222222111111\r\n"
+                           "\r\n"
+                           "KKKKKJJJJJIIIIIHHHHHGGGGGFFFFFEEEEEDDDDDCCCCCBBBBBAAAAA")),
+      AZ_OK);
+
+  assert_return_code(
+      az_http_pipeline_policy_header_validation(policies, NULL, &hrb, &response), AZ_OK);
+}
+
+static void test_az_http_pipeline_policy_header_validation_invalid(void** state)
+{
+  (void)state;
+
+  uint8_t buf[100];
+  uint8_t header_buf[(2 * sizeof(az_pair))];
+  memset(buf, 0, sizeof(buf));
+  memset(header_buf, 0, sizeof(header_buf));
+
+  az_span url_span = AZ_SPAN_FROM_BUFFER(buf);
+  az_span remainder = az_span_copy(url_span, AZ_SPAN_FROM_STR("url"));
+  assert_int_equal(az_span_size(remainder), 97);
+  az_span header_span = AZ_SPAN_FROM_BUFFER(header_buf);
+  _az_http_request hrb;
+
+  assert_return_code(
+      az_http_request_init(
+          &hrb, &az_context_app, az_http_method_get(), url_span, 3, header_span, AZ_SPAN_NULL),
+      AZ_OK);
+
+  _az_http_policy policies[1] = {            
+            {
+              ._internal = {
+                .process = test_policy_transport,
+                .p_options = NULL,
+              },
+            },
+        };
+  az_http_response response = { 0 };
+  assert_return_code(
+      az_http_response_init(
+          &response,
+          AZ_SPAN_FROM_STR("HTTP/1.1 404 Not Found\r\n"
+                           "Header11: Value11\r\n"
+                           "Header22: NNNNOOOOPPPPQQQQRRRRSSSSTTTTUUUUVVVVWWWWXXXXYYYYZZZZ\r\n"
+                           "Header33:\r\n"
```

Comment:
Add test cases for invalid white space.

Reply:
already there

Reply:
Can you point me to the test? I couldn't find a test where we have a response with whitespace  between the header and colon.

Reply:
that's part of the trimming.  He don't fail if there is spaces before or after colon.
We trim the name and value for customer instead